pylint/pyreverse/utils.py

```diff
@@ -11,10 +11,23 @@
 import shutil
 import subprocess
 import sys
+from typing import TYPE_CHECKING, Any, Callable, Optional, Tuple, Union
 
 import astroid
 from astroid import nodes
 
+if TYPE_CHECKING:
+    from pylint.pyreverse.diadefslib import DiaDefGenerator
+    from pylint.pyreverse.diagrams import ClassDiagram, PackageDiagram
+    from pylint.pyreverse.inspector import Linker
+
+    _CallbackT = Callable[
+        [nodes.NodeNG],
+        Union[Tuple[ClassDiagram], Tuple[PackageDiagram, ClassDiagram], None],
```

Comment:
Returning tuples of different lengths is probably not the best API design

Reply:
No doubt about that. The code where this happens is over 16 years old, and I haven't touched those parts of `pyreverse` yet. I can take a look at refactoring those parts that we found to be problematic due to the typing in a follow-up PR.